scala/scala.bzl

```diff
@@ -113,24 +113,27 @@ touch {statsfile}
 
 
 def _collect_plugin_paths(plugins):
-    paths = depset()
+    paths = []
     for p in plugins:
         if hasattr(p, "path"):
-            paths += [p.path]
+            paths.append(p)
         elif hasattr(p, "scala"):
-            paths += [p.scala.outputs.jar.path]
+            paths.append(p.scala.outputs.jar)
         elif hasattr(p, "java"):
-            paths += [j.class_jar.path for j in p.java.outputs.jars]
+            paths.extend([j.class_jar for j in p.java.outputs.jars])
         # support http_file pointed at a jar. http_jar uses ijar,
         # which breaks scala macros
         elif hasattr(p, "files"):
-            paths += [f.path for f in p.files if not_sources_jar(f.basename) ]
-    return paths
+            paths.extend([f for f in p.files if not_sources_jar(f.basename)])
+    return depset(paths)
 
 
 def _expand_location(ctx, flags):
   return [ctx.expand_location(f, ctx.attr.data) for f in flags]
 
+def _sort_join_path(args, sep=","):
+    return sep.join(sorted([f.path for f in args]))
```

Comment:
this sorting is not needed. Depsets have stable orderings so we don't need to sort again to achieve that.

https://docs.bazel.build/versions/master/skylark/lib/depset.html